### PR TITLE
3PC evaluation report skeleton

### PR DESCRIPTION
**Pull request by g4s8**
For #162 - started report tex sources:
 - Added abstract
 - Described the problem
 - Started solution section
 - Added todos for later
 
See [PDF](https://github.com/cqfn/degitx/files/6076102/3pc-consistency.pdf).

### CODE REVIEW

**Review by stepanvalyavskiy — DISMISSED** [review]

**Comment by g4s8** [comment]
@IlyaMoskva please check this PR

File: reports/3pc-consistency/3pc-consistency.tex [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+\acmConference{none}
+\editor{none}
+
+\begin{document}
+
+\begin{abstract}
+  For git repository replication distributed system we need to use atomic
+  commit implementation based on git reference transaction hooks (todo: add reference to previous report).
+  There are three most popular implementations: two-phase commit (2PC), three-phase commit (3PC),
+  and Paxos commit protocols. 2PC protocol is not really fault tolerant, since it may turn storage
```
Comment (IlyaMoskva) [inline reports/3pc-consistency/3pc-consistency.tex:34]: Here could be proof-link to related work/paper

File: reports/3pc-consistency/3pc-consistency.tex [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  replicas into inconsistent state in case of transaction manager (TM) failure. Paxos-commit is a fault-tolerant
+  version of transaction commit protocol. But it pays for fault-tolerance by sending much more messages via network
+  for distributing the resource manager (RM) votes: \texttt{3N - 3} messages for 2PC vs
+  \texttt{(N - 1)(2F + 3)} messages for Paxos commit co-located faster version (where N is number of nodes,
+  and F is a number of tolerated faults)
+  (todo: add references  J. Gray and L. Lamport, “Consensus on transaction commit).
+  Three phase commit (3PC) claims to be more fault-tolerant comparing to 2PC but it still have a path to become
+  inconsistent. On the other hand, it's much faster comparing to Paxos commit and uses fewer messages to perform a
```
Comment (IlyaMoskva) [inline reports/3pc-consistency/3pc-consistency.tex:42]: todo: proof-link with related research

**Review by IlyaMoskva — APPROVED** [review]
@g4s8 , thank you for this first version of the document. It's very good.

**Comment by IlyaMoskva** [comment]
@g4s8 , changes are approved and required check is successful. Please merge this PR.

File: reports/3pc-consistency/3pc-consistency.tex (hunk 1 above)
Comment (g4s8) [reply reports/3pc-consistency/3pc-consistency.tex:34]: You're right, I added a `todo` at 34 line, it was described the problem of 2PC in section 3.3 of "J. Gray and L. Lamport, Consensus on transaction commit" paper.

File: reports/3pc-consistency/3pc-consistency.tex (hunk 2 above)
Comment (g4s8) [reply reports/3pc-consistency/3pc-consistency.tex:42]: Fixed

**Review by IlyaMoskva — APPROVED** [review]

**Pull request merged**